File: src/main/kotlin/dev/jefy/connectpro/portfolio/domain/vo/Pricing.java
package dev.jefy.connectpro.portfolio.domain.vo;


import java.math.BigDecimal;
import java.util.List;

import jakarta.persistence.Embeddable;

/**
 * @author Jôph Yamba
 */
@Embeddable
public record Pricing (
        BigDecimal basePrice,
        int deliveryDays,
        boolean isNegotiable,
        List<Extra> extras 
) {}

File: src/main/kotlin/dev/jefy/connectpro/portfolio/domain/vo/Pricing.kt
package dev.jefy.connectpro.portfolio.domain.vo

import dev.jefy.connectpro.shared.domain.vo.ExtraType
import jakarta.persistence.Embeddable
import java.math.BigDecimal

@Embeddable
data class Pricing(
    var basePrice: BigDecimal,
    var deliveryDays: Int,
    var isNegotiable: Boolean,
    var extras: List<Extra>
)

data class Extra(
    val type: ExtraType,
    val name: String?,
    val price: Double,
    val description: String?
) 